pkg/asset/agent/joiner/clusterinfo.go

```diff
@@ -688,3 +691,31 @@ func (ci *ClusterInfo) reportResult(ctx context.Context) error {
 
 	return workflowreport.GetReport(ctx).StageResult(workflow.StageClusterInspection, string(data))
 }
+
+func (ci *ClusterInfo) removeDuplicates(input string) string {
+	entries := strings.Split(strings.TrimSpace(input), ",")
+
+	duplicates := []string{}
+	uniqueMap := make(map[string]bool)
+	var uniqueList []string
+	for _, entry := range entries {
+		trimmedEntry := strings.TrimSpace(entry)
+
+		// Check if the entry is not empty and hasn't been added to the map yet
+		if trimmedEntry == "" {
+			continue
+		}
+		if _, exists := uniqueMap[trimmedEntry]; !exists {
+			// If it doesn't exist, add it to the map and the unique list
+			uniqueMap[trimmedEntry] = true
+			uniqueList = append(uniqueList, trimmedEntry)
+		} else {
+			duplicates = append(duplicates, trimmedEntry)
+		}
+	}
+
+	if len(duplicates) > 0 {
+		logrus.Tracef("Duplicates detected in noProxy: %v", duplicates)
```

Comment:
I'd keep it as `logrus.Infof` level, so that the user may notice it (and eventually manually fix the Proxy)

```suggestion
		logrus.Infof("Skipping the following duplicates detected in Proxy.noProxy field: %v", duplicates)
```